android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -565,6 +565,10 @@ public void setSource(WebView view, @Nullable ReadableMap source) {
                 view.getSettings().setUserAgentString(headers.getString(key));
               }
             } else {
+              if (key.equalsIgnoreCase("Cookie")) {
+                CookieManager.getInstance().setCookie(url, headers.getString(key));
```

Comment:
Does this specifically narrows the cookie to the exact `uri` passed into the webview as the domain? Would it make more sense to actually set it to the host instead of the full uri?

Reply:
@timbrownsf Good point. We only needed it for 1 specific url in our project so it didn't occur to me to set it for a whole domain. PR updated.